Set wrong certificate path make virtual-kubelet pod crash.

---

### Environment summary

Provider : Alibaba

Version : master

K8s Master Info : ACS

Install Method : kubectl

### Issue Details

Set env in yaml for self signed cert, but the virtual-node pod crashed.
 
``` yaml
- name: APISERVER_CERT_LOCATION
          value: "/var/lib/kubelet/kubelet.crt"
- name: APISERVER_KEY_LOCATION
          value: "var/lib/kubelet/kubelet.key"
```
After trace the crash info, I found a logic bug in function `loadTLSConfig` of file `cmd/http/go`.

When error occurs, `nil` and `err` will be returned, thus assign `cancel == nil` and `retErr != nil`,  then defer statement `cancel()` be called, leading a `segmentation` crash.
``` go
func setupHTTPServer(ctx context.Context, cfg *apiServerConfig) (cancel func(), retErr error) {
	var closers []io.Closer
	cancel = func() {
		for _, c := range closers {
			c.Close()
		}
	}
	defer func() {
		if retErr != nil {
			cancel()
		}
	}()

	if cfg.CertPath == "" || cfg.KeyPath == "" {
		log.G(ctx).
			WithField("certPath", cfg.CertPath).
			WithField("keyPath", cfg.KeyPath).
			Error("TLS certificates not provided, not setting up pod http server")
	} else {
		tlsCfg, err := loadTLSConfig(cfg.CertPath, cfg.KeyPath)
		if err != nil {
			return nil, err
		}
		l, err := tls.Listen("tcp", cfg.Addr, tlsCfg)
		if err != nil {
			return nil, errors.Wrap(err, "error setting up listener for pod http server")
		}

		mux := http.NewServeMux()
		vkubelet.AttachPodRoutes(p, mux)

		s := &http.Server{
			Handler:   mux,
			TLSConfig: tlsCfg,
		}
		go serveHTTP(ctx, s, l, "pods")
		closers = append(closers, s)
	}

	if cfg.MetricsAddr == "" {
		log.G(ctx).Info("Pod metrics server not setup due to empty metrics address")
	} else {
		l, err := net.Listen("tcp", cfg.MetricsAddr)
		if err != nil {
			return nil, errors.Wrap(err, "could not setup listener for pod metrics http server")
		}

		mux := http.NewServeMux()
		vkubelet.AttachMetricsRoutes(p, mux)
		s := &http.Server{
			Handler: mux,
		}
		go serveHTTP(ctx, s, l, "pod metrics")
		closers = append(closers, s)
	}

	return cancel, nil
}
```

### Repo Steps

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b99737]

goroutine 1 [running]:
github.com/virtual-kubelet/virtual-kubelet/cmd.setupHTTPServer.func2(0xc000ac9bf0, 0xc000ac9be8)
	/go/src/github.com/virtual-kubelet/virtual-kubelet/cmd/http.go:83 +0x37
github.com/virtual-kubelet/virtual-kubelet/cmd.setupHTTPServer(0x26edf00, 0xc0008414a0, 0xc000949f00, 0x0, 0x26a1660, 0xc0008dd7a0)
	/go/src/github.com/virtual-kubelet/virtual-kubelet/cmd/http.go:100 +0x945
github.com/virtual-kubelet/virtual-kubelet/cmd.glob..func2(0x4008e40, 0xc000960ca0, 0x0, 0x2)
	/go/src/github.com/virtual-kubelet/virtual-kubelet/cmd/root.go:127 +0x4df
github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra.(*Command).execute(0x4008e40, 0xc00004c0a0, 0x2, 0x2, 0x4008e40, 0xc00004c0a0)
	/go/src/github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra/command.go:760 +0x2ae
github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x4008e40, 0x1b989be, 0x4008e40, 0xc000ac9f88)
	/go/src/github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra/command.go:846 +0x2ec
github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra/command.go:794
github.com/virtual-kubelet/virtual-kubelet/cmd.Execute()
	/go/src/github.com/virtual-kubelet/virtual-kubelet/cmd/root.go:151 +0x32
main.main()
	/go/src/github.com/virtual-kubelet/virtual-kubelet/main.go:20 +0x20
```

--- a/cmd/virtual-kubelet/commands/root/http.go
+++ b/cmd/virtual-kubelet/commands/root/http.go
@@ -57,9 +57,9 @@ func loadTLSConfig(certPath, keyPath string) (*tls.Config, error) {
 	}, nil
 }
 
-func setupHTTPServer(ctx context.Context, p providers.Provider, cfg *apiServerConfig) (cancel func(), retErr error) {
+func setupHTTPServer(ctx context.Context, p providers.Provider, cfg *apiServerConfig) (_ func(), retErr error) {
 	var closers []io.Closer
-	cancel = func() {
+	cancel := func() {
 		for _, c := range closers {
 			c.Close()
 		}

